src/frontends/tensorflow/src/op/dynamic_partition.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <limits>
+
+#include "op_table.hpp"
+#include "openvino/opsets/opset9.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::opset9;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_dynamic_partition_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"DynamicPartition"});
+    auto data = node.get_input(0);
+    auto partitions = node.get_input(1);
+    auto partitions_type = partitions.get_element_type();
+
+    // normalize partitions input since it can be a scalar or 1D tensor
+    auto new_parts_shape = make_shared<Constant>(element::i64, Shape{1}, -1);
+    auto norm_partitions = make_shared<Reshape>(partitions, new_parts_shape, true);
+
+    // retrieve num_partitions attribute
+    auto num_partitions = node.get_attribute<int64_t>("num_partitions");
+
+    // compute how many slices are collected for each partition
+    // that will be used as split_legths
+    auto start = make_shared<Constant>(partitions_type, Shape{}, 0);
+    auto end = make_shared<Constant>(partitions_type, Shape{}, num_partitions);
+    auto step = make_shared<Constant>(partitions_type, Shape{}, 1);
+    auto range_num_parts = make_shared<Range>(start, end, step, partitions_type);
+
+    auto unsqueeze_axis1 = make_shared<Constant>(element::i64, Shape{1}, 0);
```

Comment:
I see a lot of duplicated constants in the code
it's better to reuse them
auto const_zero = make_shared<Constant>(partitions_type, Shape{}, 0);
auto const_one = make_shared<Constant>(partitions_type, Shape{}, 0);

Reply:
I think for `axis` constant we need to use `int64` as it is done in the frameworks and to have as a rule for all frameworks.
The idea about sharing constant (scalar constant) is not quite well because the graph becomes to look ugly and other frameworks also do not use shared scalar constant for different operations. 
Actually, I am open to discuss this, interesting question indeed. Let us discuss with plugin teams how it laid down on hardware:)